pkg/admission/server.go

```diff
@@ -50,6 +50,16 @@ var (
 		Version:  v1alpha2.SchemeGroupVersion.Version,
 		Resource: "httproutes",
 	}
+	v1a1gatewayGVR = meta.GroupVersionResource{
+		Group:    v1alpha1.SchemeGroupVersion.Group,
+		Version:  v1alpha1.SchemeGroupVersion.Version,
+		Resource: "gateways",
+	}
+	v1a2gatewayGVR = meta.GroupVersionResource{
+		Group:    v1alpha2.SchemeGroupVersion.Group,
+		Version:  v1alpha2.SchemeGroupVersion.Version,
+		Resource: "gateways",
+	}
```

Comment:
(This is not in scope of this PR but I don't know where else to discuss this.)
Should these GVRs (and GVKs in future?) be defined in the `v1alphaX` package and be exported?
We don't have any duplication yet but I wanted to know if that is a good practice in general or is that advised against.

Reply:
This is a good question. Although it could be useful, I don't think there's any upstream precedent for doing that. For example, comparing our [generated register file](https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/zz_generated.register.go) with a corresponding [upstream register file](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/discovery/register.go) shows that we already have a nearly identical structure to upstream types files. Not saying we can't/shouldn't add to that though.

Reply:
I know in Contour we've duplicated the same to use them, so would be nice to have a central spot to reference. I'm not sure if that's a good practice or not.